tools/build.hpp

```diff
@@ -80,6 +81,22 @@ struct build_project {
    }
 
    int32_t exec() {
+      if (std::system("which cdt-cpp > /dev/null 2>&1") != 0 ){
```

Comment:
While there is currently no move to support Windows, it would be nice to be in the habit of writing portable code. I do not believe `which` is commonly installed in Windows.

Please consider the following change:
```suggestion
      if (std::system("cdt-cpp --version > /dev/null 2>&1") != 0 ){
```

If you *really* want to make it cross -platform, the command for windows needs to be piped to `null` instead of `/dev/null`. But that's more effort than I think is necessary.

Reply:
> While there is currently no move to support Windows, it would be nice to be in the habit of writing portable code. I do not believe `which` is commonly installed in Windows.
> 
> Please consider the following change:
> 
> If you _really_ want to make it cross -platform, the command for windows needs to be piped to `null` instead of `/dev/null`. But that's more effort than I think is necessary.

I was using cdt-cpp --version at the beginning :)

Reply:
I don't feel that task really worth it to work on cross platform code. doing so would require some library for piping output cross platform way